### ISSUE TITLE
Add Special Formatting to Select Input Fields 

### ISSUE THREAD

**Issue by ryanwiemer** [chrisheninger/rental-resume.com#3]
Automatically add special formatting to select form input fields:
- Currency
- Phone Number
- Zip Code

**Comment by chrisheninger** [comment]
What do you think we should do for input masks on each field? Should we worry about internationalization? Here's my first shot:

Currency:
- Only allows numbers, then we format (commas/currency symbol) on preview **or**
- Only allows numbers + commas + currency symbols, then we display exactly what the user input?

Phone Number:
- Only allows numbers, then we format (depending on length, we can wrap area code in parenthesis or add dashes, etc.) on preview **or**
- Only allows numbers + parenthesis + dashes, then we display exactly what the user input?
- There's probably a little more to this that we can think through...

Zip Code:
- Not sure what to do here, I know that a lot of countries have letters in their postal code. Will need to do some research, any thoughts/knowledge on this?

**Comment by ryanwiemer** [comment]
Regarding internationalization let's save that for a different ticket / future enhancement and focus first on the U.S to keep this simple. Here's my take on the formatting but I'm open to anything different if its more pain than it's worth

**Currency:**
- Only allows numbers + commas + $, then on preview / print regardless of how they input their number we display it in this format: **$50,000** 
- Don't allow the user to input a decimal or cents

**Phone Number:**
- Only allows numbers + parenthesis + dashes, then on preview / print regardless of how they input their number we display it in this format: **(555) 555-5555** 
- The only issue here is what to do if no area code is entered. Just ignore that?

**Zip Code:**
- Only allows numbers + dash, then on preview / print regardless of how they input their number we display it in this format: **94610**
- Even if the users enters the full zip code we should shorten it to the more common 5 digit format.

**Comment by chrisheninger** [comment]
Here's where I ended up for now, go ahead and pull the branch and `npm install` then see what you think. Squash merge the pull request and close branch if you're good with it.

Currency: Number field with 1000 stepper, formats on preview page to insert commas. The prefix $ on the form needs some styling lovin'.

Phone: Input mask (999) 999-9999. Formatting carries over to preview page.

Zip: Input mask 99999. Limited it to 5 characters.

New! State: Added select input with the 50 states + DC.
